Server/src/routes/queryDataRouter.ts

```diff
@@ -0,0 +1,13 @@
+var express = require('express');
+const getDataRouter = require('express').Router();
+const getDataController = require('../controllers/getDataController');
+
+/**
+ * This file contains the routes for a call to query or obtain a dataset. 
+ * If an API call is made to /getData then the request is routed to the getDataController
+ * to continue processing of the request.
+ */
+
+getDataRouter.get('/dataset*', getDataController.createRequestForData);
```

Comment:
We should do this a little cleaner than wild card here. Im pretty sure this lets the user submit anything at all

Reply:
The user can submit anything, but it is all filtered by getDataService. Something cleaner should be possible, as we know exactly what sort of params a user can send, as per the list below:

datasetId
material
firstName
lastName
year
categoryId 
subcategoryId

That said, I'm not entirely sure how to approach such a change outside of making 24 routes for each possible query, which is equally messy.

Reply:
We dont need 24 routes. We can make them all optional parameters but we should specify the accepted parameters

Reply:
That would work well, but I'm not sure where to start for that code wise. Do you have some documentation or other links for such?

Reply:
I realize with express we don't really need to. Since we have access to all input params with request.query.{paramName} so I suppose its ok to leave it as such